File: android/http-core-java/src/main/java/com/caldremch/http/core/abs/IConvert.java
package com.caldremch.http.core.abs;

import com.caldremch.http.core.framework.base.IBaseResp;
import com.caldremch.http.core.model.ResponseBodyWrapper;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Leon on 2022/10/30.
 */
public interface IConvert <R>{
    <T> IBaseResp<T> convert(@NotNull ResponseBodyWrapper<R> responseBodyWrapper, @NotNull Class<T> cls);
}

File: http-core/src/main/java/com/caldremch/http/core/abs/IConvert.kt
package com.caldremch.http.core.abs

import com.caldremch.http.core.model.ResponseBodyWrapper

/**
 *
 * @author Caldremch
 *
 * @date 2020-01-09 15:33
 *
 * @email caldremch@163.com
 *
 * @describe
 *
 **/
interface IConvert<R> {
    fun <T> convert(responseBodyWrapper: ResponseBodyWrapper<R>, clz:Class<T>): T
}